spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/procedures/RollbackToSnapshotProcedure.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.spark.sql.catalyst.procedures
+
+import org.apache.spark.sql.{Row, SparkSession}
+import org.apache.spark.sql.connector.catalog.TableCatalog
+import org.apache.spark.sql.execution.command.RunnableCommand
+import org.apache.spark.sql.types.{LongType, StringType}
+
+case class RollbackToSnapshotProcedure(catalog: TableCatalog) extends BaseIcebergProcedure {
+
+  override val name: String = "rollback_to_snapshot"
+  override val parameters: Seq[Parameter] = Seq(
+    RequiredParameter("table", StringType),
+    RequiredParameter("snapshot_id", LongType))
+
+  override def createRunnableCommand(args: Array[Any]): RunnableCommand = {
+    val tableName = getAs[String](args, index = 0)
+    val snapshotId = getAs[Long](args, index = 1)
+    RollbackToSnapshotCommand(catalog, tableName, snapshotId)
+  }
+}
+
+case class RollbackToSnapshotCommand(
+    catalog: TableCatalog,
+    tableName: String,
+    snapshotId: Long) extends BaseIcebergProcedureCommand {
+
+  override def run(spark: SparkSession): Seq[Row] = {
+    val table = loadTable(spark, catalog, tableName)
+
+    table.manageSnapshots()
+      .rollbackTo(snapshotId)
+      .commit()
+
+    // TODO: refresh catalog and all cached plans that contain this table?
```

Comment:
When caching, the table instance is reused across all plans and is automatically updated by the `commit`.

When not caching, I think it's debatable whether we should refresh.